### DISCUSSION TITLE
Adding a writable variable to a Kernel

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#29534] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi all,

In the `computeQpResidual` method in a kernel, I want to save an intermediate value to an aux variable. This won't affect the solve, it is just for post processing reasons.
I am having trouble properly writing in the aux variable in a way that it is writable.

For example
```
ADReal
someKernel::computeQpResidual(){

ADReal value = _grad_test[_i][_qp] * _grad_u[_qp];
_aux_var[_qp] = value

return value * (some function);

}
```
I want to write to _aux_var[_qp]. 

What is the correct way to declare this aux variable?

**Comment by GiudGiud** [comment, marked as answer]
Hello

> In the computeQpResidual method in a kernel, I want to save an intermediate value to an aux variable. This won't affect the solve, it is just for post processing reasons.

We dont typically do that. Auxvariables are computed by auxkernels. 
You can use this interface to do this and save to a tagged vector
https://mooseframework.inl.gov/source/interfaces/TaggingInterface.html

The auxvariable is then used along with a TagVectorAux to retrieve the value that has been saved in the tagged vector

**Reply by TheGreatCid** [reply]
Ah I always forget about this. Thanks!
